Mathlib/SetTheory/ZFC/Ordinal.lean

```diff
@@ -89,4 +97,42 @@ theorem isTransitive_iff_subset_powerset : x.IsTransitive ↔ x ⊆ powerset x :
 
 alias ⟨IsTransitive.subset_powerset, _⟩ := isTransitive_iff_subset_powerset
 
+/-- A set `x` is a von Neumann ordinal when it's a transitive set, that's transitive under `∈`. We
+will prove that this further implies that `x` is well-ordered under `∈`.
+
+The transitivity condition is written in an even weaker form, where `a ∈ b` and `b ∈ c` imply
+`a ∈ c` when only `c ∈ x` and not `a ∈ x` or `b ∈ x` are known a priori. -/
+def IsOrdinal (x : ZFSet) : Prop :=
+  x.IsTransitive ∧ ∀ a b c : ZFSet, a ∈ b → b ∈ c → c ∈ x → a ∈ c
+
+namespace IsOrdinal
+
+protected theorem isTransitive (h : x.IsOrdinal) : x.IsTransitive :=
+  h.1
+
+theorem subset_of_mem (h : x.IsOrdinal) : y ∈ x → y ⊆ x :=
```

Comment:
Hmm, I'm not entirely convinced this one and the one below should be lemmas since they're just combinations of `isTransitive` and lemmas about transitive sets. `mem_trans'` is fine though

Reply:
There's a bunch of lemmas about `∈` and `⊆` we can state on ordinals that aren't true for general transitive sets, such as
```lean 4
variable {x y z : ZFSet} 

example : IsOrdinal x → IsOrdinal y → x ⊆ y → y ∈ z → x ∈ z := sorry
example : IsOrdinal x → IsOrdinal y → x ⊆ y ∨ y ⊆ x := sorry
example : IsOrdinal x → IsOrdinal y → x ∈ y ∨ x = y ∨ y ∈ x := sorry
```
I think it's convenient to have these small wrappers over `IsTransitive` - that way you don't have to stop and think whether a theorem holds for all transitive sets or is exclusive to ordinals.

Reply:
Okay, fair enough